core/templates/pages/exploration-player-page/templates/lesson-information-card-modal.component.ts

```diff
@@ -276,6 +282,38 @@ export class LessonInformationCardModalComponent extends ConfirmOrCancelModal {
     });
   }
 
+  getCheckpointsCardIndexs(): number[] {
+    const checkpointCardIndexs: number[] = [];
+    const numberOfCards = this.playerTranscriptService.getNumCards();
+
+    for (let i = 0; i < numberOfCards; i++) {
+      const stateName = this.playerTranscriptService.getCard(i).getStateName();
+      const correspondingState =
+        this.explorationEngineService.getStateFromStateName(stateName);
+      if (correspondingState.cardIsCheckpoint) {
+        checkpointCardIndexs.push(i);
+      }
+    }
+    return checkpointCardIndexs;
+  }
+
+  returnToCheckpointIfCompleted(index: number): void {
+    const checkpointCardsIndex = this.getCheckpointsCardIndexs();
+    const cardIndex = checkpointCardsIndex[index];
+
+    if (cardIndex === undefined) {
+      console.error('No card index associated with this checkpoint.');
+    }
+    if (this.checkpointStatusArray[index] !== CHECKPOINT_STATUS_COMPLETED) {
+      console.error('Cannot return to an incomplete checkpoint.');
```

Comment:
I think it's not good here to log an error. Error is something bad, unexpected, a user clicking on unfinished checkpoints is not bad. Maybe no logs at all is fine.

Errors in logs draw attention of developers, this action should not draw attention

Reply:
Now it only returns, instead of logging an error. Thank you.